### PR TITLE
Add download-artifact action

### PR DESCRIPTION
**Pull request by jonchang**
Imported via git-subtree from https://github.com/dawidd6/action-download-artifact

This is needed to download artifacts for Homebrew/core and Linuxbrew/core

cc @dawidd6

Licensing:

image

### CODE REVIEW

File: download-artifact/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 Dawid Dziurla
```
Comment (MikeMcQuaid) [inline download-artifact/LICENSE]: @dawidd6 Could this be updated/relicensed to Homebrew (and perhaps be the same license as Homebrew/brew?)

Comment (dawidd6) [reply download-artifact/LICENSE]: Sure, just take it :smile:

File: download-artifact/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+An action that downloads and extracts uploaded artifact associated with given workflow and commit.
+
+Let's suppose you have a workflow with a job in it that at the end uploads an artifact using `actions/upload-artifact` action and you want to download this artifact in another workflow that is run after the first one. Official `actions/download-artifact` does not allow this. That's why I decided to create this action. By knowing only the workflow name and commit SHA, you can download the previously uploaded artifact from different workflow associated with that commit and use it.
+
+## Usage
+
+```yaml
+- name: Download artifact
+  uses: dawidd6/action-download-artifact@master
```
Comment (MikeMcQuaid) [inline download-artifact/README.md]: Might want to update this to Homebrew/action...

File: download-artifact/action.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: Download workflow artifact
+description: Download and extract an artifact associated with given workflow and commit
+author: dawidd6
+branding:
+  icon: download
+  color: blue
+inputs:
+  github_token:
+    description: Github token
```
Comment (MikeMcQuaid) [inline download-artifact/action.yml]:
```suggestion
    description: GitHub token
```

File: download-artifact/action.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    required: true
+  workflow:
+    description: Workflow name
+    required: true
+  pr:
+    description: Pull request number
+    required: false
+  commit:
+    description: Commit SHA
```
Comment (MikeMcQuaid) [inline download-artifact/action.yml]:
```suggestion
    description: Commit SHA1
```

File: download-artifact/README.md (hunk 2 above)
Comment (jonchang) [reply download-artifact/README.md]: Fixed in 86f63e9f8b7ed8b206aa30df0af625a2e3ac7771

**Comment by jonchang** [comment]
`git-ssh` test was fixed on `master` in b762f37682d1988e1e2c3279a70b1cfdb22de20c

Something weird is happening with `git rebase` and I can't fix up the old commits. I'll fix the test for `download-artifact` after this pull request is merged.

**Comment by jonchang** [comment]
Should this be squash and merged or have a merge commit created (to preserve git history)?

**Comment by MikeMcQuaid** [comment]
> Should this be squash and merged or have a merge commit created (to preserve git history)?

I have no preference: up to you!

**Comment by jonchang** [comment]
Merged in 2073250e769b98c522223311e7a6fd96d70e8492

**Comment by jonchang** [comment]
Thanks all!

**Pull request closed without merging**